cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -17,23 +17,36 @@ limitations under the License.
 package cluster
 
 import (
+	"context"
 	"time"
 
 	"github.com/pkg/errors"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
 	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
 	manifests "sigs.k8s.io/cluster-api/cmd/clusterctl/config"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/util"
 	logf "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
 	utilresource "sigs.k8s.io/cluster-api/util/resource"
 	utilyaml "sigs.k8s.io/cluster-api/util/yaml"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	// This is declared as a variable mapping a version number to the hash of the
+	// embedded cert-manager.yaml file.
+	// The hash is used to ensure that when the cert-manager.yaml file is updated,
+	// the version number marker here is _also_ updated.
+	// If the cert-manager.yaml asset is modified, this line **MUST** be updated
+	// accordingly else future upgrades of the cert-manager component will not
+	// be possible, as there'll be no record of the version installed.
+	embeddedCertManagerManifestVersion = map[string]string{"v0.15.2": "56bd9851567be27698114a73625594a9"}
```

Comment:
Wondering if we should split this into two labels, so we can rely on a well know label name that will simplify the implementation of the upgrade logic eg.
```
clusterctl.cluster.x-k8s.io/version: "v0.15.2"
clusterctl.cluster.x-k8s.io/hash: "56bd9851567be27698114a73625594a9"
```